### PR TITLE
Refactor Edit Categories

### LINKED ISSUE #299: Refactor: Edit Categories

### PR DESCRIPTION
**Pull request by stevefosterwta**
Closes Github issue 299

# Description of Work
Refactored EditCategoriesViewController 

## Notes to Test
The changes in this PR assumes the following feature flag configuration:
- `EditPhrasesEnabled` is enabled 
- `RefactoredInterfaceEnabled` is enabled 

**Screens updated:** 
There was an update to the GazeableButton to better handled all the states a button could be in (normal, selected, highlighted and disabled). 

- Category: the up/down buttons should display their disabled state.
- Category Detail: disabled buttons are distinguishable from active buttons. 
- The navigation bar: Mentioned since the GazeableButton was update and the nav bar uses GazeableButton. 
- The paging buttons: Same as the navigation bar.

### CODE REVIEW

File: Vocable.xcodeproj/xcshareddata/xcschemes/Development.xcscheme [hunk 1]
```diff
@@ -93,7 +93,7 @@
          <EnvironmentVariable
             key = "RefactoredInterfaceEnabled"
             value = ""
-            isEnabled = "NO">
+            isEnabled = "YES">
```
Comment (Clstroud) [inline Vocable.xcodeproj/xcshareddata/xcschemes/Development.xcscheme]: I'm not sure this should be on by default yet?

**Comment by stechkev** [comment]
Issue:
When gazing at the "show category", if the cursor is highlighting the toggle icon itself then the yellow highlighting around the button is no longer shown

File: Vocable/Common/Views/GazeableButton.swift [hunk 2]
```diff
@@ -43,32 +44,32 @@ class GazeableButton: UIButton {
...
+//        get {
+//            return fillColor(for: .normal)
+//        }
+//        set {
+//            print("Warning: using deprecated `fillColor` property on GazeableButton")
+//            for state in defaultIBStates {
+//                setFillColor(newValue, for: state)
+//            }
+//        }
+//    }
```
Comment (Clstroud) [inline Vocable/Common/Views/GazeableButton.swift]:
```suggestion
```

File: Vocable/Common/VocableNavigationBar.swift [hunk 3]
```diff
@@ -93,7 +93,7 @@ private class ContentView: UIView {
             titleCenterY.priority = .init(rawValue: 999)
             constraints += [
                 titleCenterX,
-                titleCenterY,
+                titleCenterY
 //                titleView.leadingAnchor.constraint(lessThanOrEqualTo: layoutMargins.leadingAnchor),
 //                titleView.trailingAnchor.constraint(greaterThanOrEqualTo: layoutMargins.trailingAnchor)
```
Comment (Clstroud) [inline Vocable/Common/VocableNavigationBar.swift:100]: We should add a warning to revisit this. I think long strings will be a problem otherwise.

**Review by Clstroud — APPROVED** [review]
💯

File: Vocable/Common/Views/GazeableButton.swift (hunk 2 above)
Comment (stevefosterwta) [reply Vocable/Common/Views/GazeableButton.swift]: @Clstroud Should we remove these deprecated functions?

**Pull request merged**